api/server/router/volume/volume_routes.go

```diff
@@ -24,18 +52,62 @@ func (v *volumeRouter) getVolumesList(ctx context.Context, w http.ResponseWriter
 	if err != nil {
 		return err
 	}
+
+	version := httputils.VersionFromContext(ctx)
+	if versions.GreaterThanOrEqualTo(version, clusterVolumesVersion) {
+		clusterVolumes, swarmErr := v.cluster.GetVolumes(types.VolumeListOptions{Filters: filters})
+		if swarmErr != nil && !isNoSwarmErr(swarmErr) {
+			// if there is a swarm error, we may not want to error out right
+			// away. the local list probably worked. instead, let's do what we
+			// do if there's a bad driver while trying to list: add the error
+			// to the warnings. don't do this if swarm is not initialized.
+			warnings = append(warnings, swarmErr.Error())
+		}
+		// add the cluster volumes to the return
+		volumes = append(volumes, clusterVolumes...)
+	}
```

Comment:
Given that `v.cluster` is actually the whole cluster backend, one option could be to add `IsManager() bool` to the `volume.ClusterBackend` interface. That way we can check here if the current node is a Swarm manager, and otherwise skip this block; that way, we don't have to use the (rather hacky) `isNoSwarmErr()` handling here.

```suggestion
	version := httputils.VersionFromContext(ctx)
	if versions.GreaterThanOrEqualTo(version, clusterVolumesVersion) && v.cluster.IsManager() {
		clusterVolumes, err := v.cluster.GetVolumes(types.VolumeListOptions{Filters: filters})
		if err != nil {
			// if there is a swarm error, we may not want to error out right
			// away. the local list probably worked.
			warnings = append(warnings, err.Error())
		}
		volumes = append(volumes, clusterVolumes...)
	}
```